src/Tabs/TabTemplate.js

```diff
@@ -1,29 +1,28 @@
-import React, {Component, PropTypes} from 'react';
+import React, {PropTypes} from 'react';
 
-class TabTemplate extends Component {
-  static propTypes = {
-    children: PropTypes.node,
-    selected: PropTypes.bool,
-  };
+const TabTemplate = ({children, selected, templateStyles}) => {
+  if (!selected) {
+    styles.height = 0;
+    styles.overflow = 'hidden';
+  }
 
-  render() {
-    const styles = {
-      width: '100%',
-      position: 'relative',
-      textAlign: 'initial',
-    };
+  return (
+    <div style={Object.assign(styles, templateStyles)}>
+      {children}
+    </div>
+  );
+};
 
-    if (!this.props.selected) {
-      styles.height = 0;
-      styles.overflow = 'hidden';
-    }
+TabTemplate.propTypes = {
+  children: PropTypes.node,
+  selected: PropTypes.bool,
+  templateStyles: PropTypes.object,
+};
 
-    return (
-      <div style={styles}>
-        {this.props.children}
-      </div>
-    );
-  }
-}
+const styles = {
```

Comment:
I would rather have this block defined before it's used.
We have an eslint rule (coming from Airbnb) preventing this pattern.
I think it's simpler for understanding the source-code (predictability).